src/sentry/services/hybrid_cloud/project/impl.py

```diff
@@ -0,0 +1,27 @@
+from __future__ import annotations
+
+from sentry.models import Project, ProjectOption
+from sentry.services.hybrid_cloud import OptionValue
+from sentry.services.hybrid_cloud.project import ProjectService, RpcProject, RpcProjectOptionValue
+
+
+class DatabaseBackedProjectService(ProjectService):
+    def get_option(self, *, project: RpcProject, key: str) -> RpcProjectOptionValue:
+        from sentry import projectoptions
+
+        orm_project = Project.objects.get(id=project.id)
```

Comment:
```suggestion
        orm_project = Project.objects.get_from_cache(id=project.id)
```

We should be able to read from cache here as reading/writing options doesn't use much project state.